tests/components/icloud/test_config_flow.py

```diff
@@ -0,0 +1,307 @@
+"""Tests for the iCloud config flow."""
+import logging
+from unittest.mock import patch, MagicMock, PropertyMock
+import pytest
+
+from pyicloud.exceptions import PyiCloudFailedLoginException
+
+from homeassistant import data_entry_flow
+from homeassistant.components.icloud import config_flow
+from homeassistant.components.icloud.config_flow import (
+    CONF_TRUSTED_DEVICE,
+    CONF_VERIFICATION_CODE,
+)
+from homeassistant.components.icloud.const import (
+    DOMAIN,
+    CONF_ACCOUNT_NAME,
+    CONF_GPS_ACCURACY_THRESHOLD,
+    CONF_MAX_INTERVAL,
+    DEFAULT_GPS_ACCURACY_THRESHOLD,
+    DEFAULT_MAX_INTERVAL,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+USERNAME = "username@me.com"
+PASSWORD = "password"
+ACCOUNT_NAME = "Account name 1 2 3"
+ACCOUNT_NAME_FROM_USERNAME = None
+MAX_INTERVAL = 15
+GPS_ACCURACY_THRESHOLD = 250
+
+TRUSTED_DEVICES = [
+    {"deviceType": "SMS", "areaCode": "", "phoneNumber": "*******58", "deviceId": "1"}
+]
+
+
+@pytest.fixture(name="session")
+def mock_controller_session():
+    """Mock a successful session."""
+    with patch("pyicloud.base.PyiCloudSession"):
+        yield
+
+
+@pytest.fixture(name="service")
+def mock_controller_service():
+    """Mock a successful service."""
+    with patch("pyicloud.base.PyiCloudService") as service_mock:
```

Comment:
Here's an example how to patch a class:
https://github.com/MartinHjelmare/home-assistant/blob/a0397f46f88f0cec99a2f1f9476d5ef30aee05b7/tests/components/mysensors/conftest.py#L91

https://github.com/MartinHjelmare/home-assistant/blob/a0397f46f88f0cec99a2f1f9476d5ef30aee05b7/tests/components/mysensors/common.py#L38-L56

In the case here we may want to create the mocked instance as a local variable so we can yield it. So instead of using `side_effect`, create the mocked instance with set attributes and set it as `return_value` in the patch. Then yield the mocked instance too.